src/Raven.Server/Storage/Layout/StorageLoader.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using Raven.Server.Config;
+using Raven.Server.Documents;
+using Sparrow.Platform;
+using Voron;
+using Voron.Exceptions;
+using Voron.Util.Settings;
+
+namespace Raven.Server.Storage.Layout
+{
+    public static class StorageLoader
+    {
+        public static StorageEnvironment OpenEnvironment(StorageEnvironmentOptions options, StorageEnvironmentWithType.StorageEnvironmentType type)
+        {
+            try
+            {
+                if (options is StorageEnvironmentOptions.DirectoryStorageEnvironmentOptions directoryOptions)
+                    return OpenEnvironmentWithPossibleLayoutUpdate(directoryOptions, type);
+
+                return new StorageEnvironment(options);
+            }
+            catch (Exception)
+            {
+                options.Dispose();
+
+                throw;
+            }
+        }
+
+        private static StorageEnvironment OpenEnvironmentWithPossibleLayoutUpdate(StorageEnvironmentOptions.DirectoryStorageEnvironmentOptions options, StorageEnvironmentWithType.StorageEnvironmentType type)
+        {
+            try
+            {
+                var oldOwnsPager = options.OwnsPagers;
+                options.OwnsPagers = false;
+                try
+                {
+                    return new StorageEnvironment(options);
+                }
+                finally
+                {
+                    options.OwnsPagers = oldOwnsPager;
+                }
+            }
+            catch (InvalidJournalException e)
+            {
+                var basePath = options.BasePath;
+                var preRtmJournalPath = basePath.Combine("Journal");
+                var journalsPath = options.JournalPath;
+                if (Directory.Exists(preRtmJournalPath.FullPath))
+                {
+                    TryMoveJournals(preRtmJournalPath, journalsPath);
+                    Directory.Delete(preRtmJournalPath.FullPath);
+                }
+                else
+                {
+                    if (TryMoveJournals(basePath, journalsPath) == false)
```

Comment:
When would this ever be triggered because of the missing journal?
You may be missing journal 16, but 17 is there, so it wouldn't call it, no?

Reply:
The existing code, in particular, `TryMoveJournals` was created to handle the change when we moved journals files to `Journals` directory in the past.

If journals are already in `Journals` directory then we're sure that missing 16 will cause throwing exception here.